bin/picca_compute_fvoigt_hcd.py

```diff
@@ -0,0 +1,142 @@
+
+    profile_comov_dist = np.interp(lin_spaced_comov_dist, comov_dist, profile_comov_dist)
+
+    return lin_spaced_comov_dist, profile_comov_dist
+
+
+def fft_profile(profile, dx):
+    """Compute Fourier transform of a voigt profile
+
+    Parameters
+    ----------
+    profile : array
+        Input voigt profile in real space (function of comoving distance)
+    dx : float
+        Comoving distance bin size
+
+    Returns
+    -------
+    (array, array)
+        (wavenumber grid, voigt profile in Fourier space)
+    """
+    # not normalized
+    size = profile.size
+    ft_profile = dx * np.fft.rfft(profile - 1)
+    k = np.fft.rfftfreq(size, dx) * (2 * np.pi)
+
+    return k, np.abs(ft_profile)
+
+
+def main():
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+                                     description=('Compute FVoigt profile'))
+
+    parser.add_argument('-o', '--output', type=str, default=None,
+                        help=('Name and path of file to write FVoigt profile to'))
+
+    parser.add_argument('--weights', type=str, default=None,
+                        help=('File with total weight as a function of observed wavelength'))
+
+    parser.add_argument('--cddf', type=str, default=None,
+                        help=('File with column density distribution function (CDDF)'))
+
+    parser.add_argument('--dla-prob', type=str, default=None,
+                        help=('File with probability of finding DLAs in a forest '
+                              'as a function of observed wavelength'))
+
+    parser.add_argument('--z-dla', type=float, default=None,
+                        help=('Mean DLA redshift'))
+
+    parser.add_argument('--normalize', action='store_true', required=False,
+                        help=('Whether the output FVoigt function is normalized'))
+
+    parser.add_argument('--positive-fvoigt', action='store_true', required=False,
+                        help=('Whether the output FVoigt function should be positive'))
+
+    args = parser.parse_args()
+
+    cddf_NHI, dN_NHI = np.loadtxt(args.cddf)
+    weights_wave = np.loadtxt(args.weights)
+    dla_prob_wave = np.loadtxt(args.dla_prob)
+
+    fidcosmo = constants.Cosmo(Om=0.3147)
+
+    comov_dist_prob, dla_prob_comov_dist = profile_wave_to_comov_dist(
+        dla_prob_wave[0], dla_prob_wave[1], fidcosmo, differential=True)
+    comov_dist_weights, weights_comov_dist = profile_wave_to_comov_dist(
+        weights_wave[:, 0], weights_wave[:, 1], fidcosmo)
+
+    weight_interp = np.interp(
+        comov_dist_prob, comov_dist_weights, weights_comov_dist, left=0, right=0)
+    mean_density = np.average(dla_prob_comov_dist, weights=weight_interp)
+
+    wave = np.arange(2000, 8000, 1)  # TODO this grid may be too sparse
+    integrand = np.empty((dN_NHI.size, wave.size // 2 + 1))
+    for i, NHI in enumerate(dN_NHI):
```

Comment:
is it correct, that in this case we want to iterate over DLAs with different column densities at the same redshift? I.e. essentially integrating the absorption of many absorbers at the same point according to some cddf? In which case this is the correct thing to do.
But it could be more clearly documented what exactly this script is supposed to be doing...

Reply:
ok, cleared this with @moonlovist and the z is just an eff. redshift where all DLAs are placed